src/System.Text.Json/src/System/Text/Json/JsonReader.cs

```diff
@@ -107,17 +110,16 @@ public JsonValueType GetJsonValueType()
                 return JsonValueType.Null;
             }
 
-            if (nextByte == '"')
+            if (nextByte == '-' || (nextByte >= '0' && nextByte <= '9'))
             {
-                return JsonValueType.String;
+                return JsonValueType.Number;
             }
 
-            throw new ArgumentOutOfRangeException();
+            throw new FormatException("Invalid json, tried to read char '" + nextByte + "'.");
         }
 
-        public Utf8String GetValue()
+        public Utf8String GetValue(JsonValueType type)
```

Comment:
I know we discussed it, but now that I look at this, I think this API turned out to be very confusing. The user will have no idea why they need to pass the type given they get a string back.